contracts/Uniswap/UniswapLib.sol

```diff
@@ -73,7 +72,7 @@ library FullMath {
         // Factor powers of two out of denominator
         // Compute largest power of two divisor of denominator.
         // Always >= 1.
-        uint256 twos = -denominator & denominator;
+        uint256 twos = denominator & (~denominator + 1);
```

Comment:
Why did these change? Aren't they from the Uniswap contracts?

Reply:
Yes, they are. But Uni contracts were originally using Solidity 0.7. Solidity 0.8 was complaining about this (negating a uint) so I had to change it.